### ISSUE TITLE
clean up `optional-state.rst`, make outref from level1 point to website

### ISSUE THREAD

**Issue by dkg** [autocrypt/autocrypt#238]
`doc/optional-state.rst` is now out of date given the overhaul to the peers table, but it could be brought back into shape. 

I'm reluctant to cut out all references to it from the spec because i want to honor the work that @BjarniRunar put into it, and to encourage more experimentation like this. 

Perhaps we can designate  https://autocrypt.org/en/latest/optional-state.html as a permanent location for this kind of "living" experimentation and forward-looking proposals that aren't yet merged into the main spec, and change `level1.rst` to refer back to the website itself instead of using a `:doc:` reference?

Blocks: #225

**Comment by azul** [comment]
I rephrased `optional-state.rst` to match the current way we update peer state in #257. The translation is not straight forward though. So I opended #258 to remind us to think it through again.

**Comment by dkg** [comment]
I don't think this is resolved because level1.rst still refers to `:doc:optional-state` in two places.

Those places should be replaced by an explicit link to https://autocrypt.org/en/latest/optional-state.html
